frontend/packages/shared/src/components/GiteaHeader/ThreeDotsMenu/LocalChangesModal/DeleteModal/DeleteModal.tsx

```diff
@@ -19,29 +21,39 @@ export type DeleteModalProps = {
  *
  * @property {boolean}[isOpen] - If the modal is open or not
  * @property {function}[onClose] - Function to execute on close
- * @property {function}[onDelete] - Function to execute on click delete
- * @property {string}[appName] - The name of the app to delete changes on
+ * @property {string}[app] - The app
+ * @property {string}[org] - The org
  *
  * @returns {ReactNode} - The rendered component
  */
-export const DeleteModal = ({
-  isOpen,
-  onClose,
-  onDelete,
-  appName,
-}: DeleteModalProps): ReactNode => {
+export const DeleteModal = ({ isOpen, onClose, app, org }: DeleteModalProps): ReactNode => {
   const { t } = useTranslation();
 
+  const { mutate: deleteLocalChanges } = useResetRepositoryMutation(org, app);
+
   const [nameToDelete, setNameToDelete] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
 
   const handleClose = () => {
     setNameToDelete('');
     onClose();
   };
 
-  const handleDelete = () => {
+  const reset = () => {
+    setIsLoading(false);
     setNameToDelete('');
-    onDelete();
+    onClose();
+  };
+
+  const handleDelete = () => {
+    setIsLoading(true);
+    deleteLocalChanges(undefined, {
+      onSuccess: () => {
+        reset();
+        toast.success(t('local_changes.modal_deleted_success'));
+      },
+      onError: () => setIsLoading(false),
+    });
```

Comment:
Ref. previous comment we can now clean this up to be something like the following code snippet:

```suggestion
  const handleDelete = (): void => {
    deleteLocalChanges(undefined, {
      onSuccess: () => {
        toast.success(t('local_changes.modal_deleted_success'));
        onClose();
      },
    });
```
I don't think we need to reset the form, since we are closing the dialog on success and then the components unmounts. What do you think? 😊

Reply:
Actually, we need to reset the form. 
Otherwise the old typed value will not be removed. I tried it now, and when I open the modal again, the value is still there.